src/test/java/com/example/java/test/junior/developer/question/QuestionServiceJunitTest.java

```diff
@@ -0,0 +1,77 @@
+package com.example.java.test.junior.developer.question;
+
+import com.example.java.test.junior.developer.dto.QuestionDto;
+import com.example.java.test.junior.developer.mapper.QuestionMapper;
+import com.example.java.test.junior.developer.model.Question;
+import com.example.java.test.junior.developer.repository.QuestionRepository;
+import com.example.java.test.junior.developer.service.QuestionService;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+public class QuestionServiceJunitTest {
+    private final QuestionRepository questionRepository = Mockito.mock(QuestionRepository.class);
+    private final QuestionMapper questionMapper = new QuestionMapper();
+    private final QuestionService questionService = new QuestionService(questionRepository, questionMapper);
+
+    @Test
+    public void createQuestion_ValidDto_ReturnsDtoWithId() {
+        QuestionDto questionDto = new QuestionDto(1, "What is polymorphism?");
+        Question question = Question.builder()
+                .id(1)
+                .name(questionDto.getName())
+                .build();
+
+        when(questionRepository.save(any(Question.class))).thenReturn(question);
```

Comment:
You need to mock you mapper - then you will add mock behavior:
`when(mapper.toEntity(questionDto)).thenReturn(question);`
Then you can get rid of `any(Question.class)` - using "any" in your mocks - is a bad sign.
You know for sure that repository will be called with your question:
`when(questionRepository.save(question).thenReturn(question);`
Finally, you need to add mock behavior for toto call.